---
title: "03_describe.qmd"
editor: visual
format:
  html:
    embed-resources: true
author: "Sarah Rosenberg Asmussen (s194689),	Mette BÃ¸ge Pedersen (s194679),	Caroline Amalie Bastholm Jensen (s213427), Jaime Noguera Piera (s233773), Yassine Turki (s231735)"
---

## Data description

Data is available here: [NCBI](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE31684)

The article published based on the data is available here: [PMC](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3569085/)

**The title of the article is:**

*Combination of a novel gene expression signature with a clinical nomogram improves the prediction of survival in high-risk bladder cancer*

A nomogram, also called an alignment chart, uses biologic and clinical variables, such as tumor grade and patient age, to graphically depict a statistical prognostic model that generates a probability of a clinical event, such as cancer recurrence or death, for a given individual([1](https://newsnetwork.mayoclinic.org/discussion/what-the-heck-is-a-nomogram-and-other-medical-research-questions-answered/)).

Urothelial carcinoma of the urinary bladder is the fifth most common cancer in the Western World, and represents 3% of cancers diagnosed globally (1).

#### About the data we use

Cancer gene expression data from 93 patients undergoing radical cystectomy (RC) between 1993 and 2004. There are 15 patients with non-MI disease on final pathologic analysis after RC, four had MI disease histologically in tissue obtained at the time of prior transurethral resection (TUR), and the remaining patients had high-risk features (extensive volume of disease, recurrent or BDG-refractory disease).

Lymph node dissection was performed in 77 patients; no patient has metastatic disease at the time of RC. Chemotherapy was administered to 3 patients as neoadjuvant, 16 patients as adjuvant, and 19 patients as salvage for recurrent disease. Case selection was restricted to those with frozen specimens with measurable volume of malignancy and adequate percentage of tumor.

*Neoadjuvant* chemo means the patient was given therapeutic agents before the main treatment.

*Adjuvant* chemo means the therapeutic agents were given after main treatment.

*Salvage* treatment is when the disease does not respond to all other treatments tried, sort of as a last attempt, and is often used for patients with advanced cancer. In our case, it seems to be the patients with recurrent bladder cancer.

#### What they did in the article

The data that we have is a microarray from 93 bladder cancer patients managed by radical cystectomy to determine gene expression patterns associated with clinical and prognostic variables. They compared their results with published bladder cancer microarray datasets comprising of 578 additional patients, and with 49 published gene signatures from multiple cancer types.

### Metadata abbreviations explained

-   MI = muscle invasive disease (in stages T2-T4)

-   Non-MI disease (in stages Ta/T1/Tis)

-   DOD = Death of disease

-   DOC = Death of other cause

-   NED = No evidence of disease - I believe this means alive and no bladder cancer

**Note: DOD, DOC and NED was after the data collection period was over**

-   CIS = Carcinoma in situ

-   TCC = Transitional Cell Carcinoma

-   RC = Radical cystectomy

## Data load

```{r}
#| warning: false
#| output: false
data_clean_wide <- read_tsv("../data/02_dat_clean_wide.tsv",
                            show_col_types = FALSE)

data_clean_wide
```

### 

## Table visualizing meta data

```{r}
table_data <- data_clean_wide |> 
  mutate(Metastasis = factor(Metastasis)) |> 
  table1(x = formula(~ Smoking + Metastasis), 
         data = _,
         caption = "Overview of meta data")

table_data
```

```{r}
# this doesn't work
ggsave(
  filename = "../results/04_table_data.pdf",
  plot = table_data,
  device = "pdf")

print(table_data)

dev.off()
```

## Counting number of genes

```{r}
data_clean_wide |> 
  select(c(GEO, ends_with("_at"))) |> 
  pivot_longer(cols = -GEO,
               names_to = "Genes",
               values_to = "Gene_expression") |> 
  distinct(Genes) |> 
  count()
```
